src/main/java/org/zalando/nakadi/service/subscription/zk/NewZkSubscriptionClient.java

```diff
@@ -0,0 +1,216 @@
+package org.zalando.nakadi.service.subscription.zk;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import static com.google.common.base.Charsets.UTF_8;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.zookeeper.KeeperException;
+import org.zalando.nakadi.domain.EventTypePartition;
+import org.zalando.nakadi.exceptions.NakadiRuntimeException;
+import org.zalando.nakadi.exceptions.runtime.MyNakadiRuntimeException1;
+import org.zalando.nakadi.service.subscription.model.Partition;
+import org.zalando.nakadi.view.SubscriptionCursorWithoutToken;
+
+/**
+ * Subscription client that uses topology as an object node and separates changeable data to a separate zk node.
+ * The structure of zk data is like this:
+ * <pre>
+ *- nakadi
+ * +- locks
+ * | |- subscription_{subscription_id}      // Node that is being used to guarantee consistency of subscription data
+ * |
+ * +- subscriptions
+ *   +- {subscription_id}
+ *     |- state                             // Node for initialization finish tracking.
+ *     |- cursor_reset                      // Optional node, presence of which says that cursor reset is in progress
+ *     |                                    // During cursor reset process no new streams can be created. At the same
+ *     |                                    // time all existing streaming sessions are being terminated.
+ *     |- sessions                          // Node contains list of active sessions
+ *     | |- {session_1}                     // Ephemeral node of session_1
+ *     | |- ....
+ *     | |- {session_N}                     // Ephemeral node of session_N
+ *     |
+ *     |- topology                          // Persistent node that holds all assignment information about partitions
+ *     |
+ *     |- offsets                           // Node that holds up all the dynamic data for this subscription (offsets)
+ *       |- {event_type_1}
+ *       | |- 0                             // Nodes that contain view of offset (data from
+ *       | |- ...                           // {@link SubscriptionCursorWithoutToken#getOffset()})
+ *       | |- {event_type_1_max_partitions} // for specific (EventType, partition) pair.
+ *       |
+ *       |- ...
+ *       |
+ *       |- {event_type_M}
+ *         |- 0
+ *         |- ...
+ *         |- {event_type_M_max_partitions}
+ *
+ *  </pre>
+ *  In topology node there it json serialized object {@link Topology}
```

Comment:
Because it naturally belongs to this, it describes the structure of the topology node you defined in this beautiful tree. Also offsets have description of the structure it stores right in the tree.